### PR TITLE
feat: implement a p + epsilon attack

### PR DESCRIPTION
**Pull request by dannybabbev**
As described in #4 and #5 

Please take a review of this code and leave your comments. It may contain bugs, as the logic is complicated.

I needed to add Kleros.sol in order to import it properly

### CODE REVIEW

**Comment by clesaege** [comment]
No need to add Kleros.sol, you can install the package.

File: contracts/p-plus-epsilon.sol [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    if (court.currentRuling(disputeID) != desiredOutcome){
+      // Calculate the redistribution amounts.
+      uint amountShift = court.getStakePerDraw();
+      uint winningChoice = court.getWinningChoice(disputeID, appeals);
+
+      // Rewards are calculated as per the one shot token reparation.
+      for (uint i=0; i < appeals; i++){ // Loop each appeal and each vote.
+
+        // Note that we don't check if the result was a tie becuse we are getting a funny compiler error: "stack is too deep" if we check.
+        // TODO: Account for ties
```
Comment (clesaege) [inline contracts/p-plus-epsilon.sol]: TODO

File: contracts/p-plus-epsilon.sol [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+              if (voteRuling == desiredOutcome){ // If the juror voted as we desired.
+                // Transfer this juror back the penalty.
+                withdraw[voteAccount] += amountShift;
+                remainingWithdraw += amountShift;
+              }
+            } else {
+              nbCoherent++;
+            }
+          }
+          if (nbCoherent > 0){
```
Comment (clesaege) [inline contracts/p-plus-epsilon.sol]: I don't understand what you are doing there.
You need to give (amountShift+epsilon) * amountDraw to everyone who voted for desiredOutcome if desiredOutcome is not winning.

Comment (clesaege) [reply contracts/p-plus-epsilon.sol]: Ok, in addition you add what they would have won from incoherent voters would they have been. So this seems correct.

**Comment by dannybabbev** [comment]
I have added some tests. It will need more though. I am continuing to work on it.

**Comment by clesaege** [comment]
Ok, when do you think it will be finished?

**Pull request merged**